### PR TITLE
Issue #227 | New Document: Database Rundown

### PR DESCRIPTION
**Pull request by ghost**
## Description
This document runs down the database physically using Fischer's original abstract ER as guide.

## Motivation and Context
The Iridium database needs a rundown for proper configuration. 

## How has this been tested?
I have checked definitions of entities with Josh for accuracy and based all ER diagrams directly off the database's physical schema.

Keep in mind this document is open for additions if there are parts I have missed. If you know something about the database that isn't present/is inaccurate and think should be added/changed, let me know here or on the discord.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: site/docs/1.-Overview/iridium-database.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+In this document, the physical structure of the database will be laid out for both cursory reference and as a rough guide for developers aiming to reconfigure the system.
+
+## Abstract Overview
+![This is the Primary Abstract Entity Relationship Diagram. There are six entities, defined below. The relationships are as follows: Managing Identity has a many-to-many relationship with Tenant. Customer Identity has a many relationship to one Tenant. Application has a many relationship to one tenant, and many-to-many with Customer Identities. External Identity Providers and Application Type both have a many relationship to one Application.](../images/iridiumdb-overview.png)
+
+<sup>Figure 1: Abstract iridium database breakdown.</sup> 
+
+Iridium uses a fairly linear data flow between its entities to achieve tasks. A third party `external identity provider` such as google or github is linked to an `application` secured with Iridium. Within a `tenant` container, `identities` of users are verified using Iridium. Within each `tenant` are `customer identities` (end-users) and `managing identities` (tenant-level system administrators).
```
Comment (joshfischer1108) [inline site/docs/1.-Overview/iridium-database.md:13]: should be Google or GitHub?

File: site/docs/1.-Overview/iridium-database.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+# Iridium Database
+
+The Iridium database stores the data and metadata for an Iridium instance. Each Iridium instance uses its own database, which is hosted in a separate container to the primary application. The database can be treated as a simple repository rather than an independent data management system.
+
+In this document, the physical structure of the database will be laid out for both cursory reference and as a rough guide for developers aiming to reconfigure the system.
+
+## Abstract Overview
+![This is the Primary Abstract Entity Relationship Diagram. There are six entities, defined below. The relationships are as follows: Managing Identity has a many-to-many relationship with Tenant. Customer Identity has a many relationship to one Tenant. Application has a many relationship to one tenant, and many-to-many with Customer Identities. External Identity Providers and Application Type both have a many relationship to one Application.](../images/iridiumdb-overview.png)
+
+<sup>Figure 1: Abstract iridium database breakdown.</sup>
```
Comment (joshfischer1108) [inline site/docs/1.-Overview/iridium-database.md:11]: Should be Iridium?

File: site/docs/1.-Overview/iridium-database.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+* **[login_descriptors]:** Describes what the login page holds for each tenant.
+* **`[tenants]:`** Stores defining data for a tenant container. Tenant names are used in the tenant's login URL, so tenant names in the database must be URL safe.
+---
+### External Identity Providers
+![This is an External Identity Provider Entity Relationship Diagram. Four entities are present, three of which are defined in the list below, one of which is defined in the tenants section. The relationships are as follows: external_identity_provider_parameter_templates has a many relationship with external_identity_provider_templates. external_identity_providers has a many relationship with external_identity_provider_templates. external_identity_providers also has a many relationship with tenants.](../images/iridiumdb-external-identity-providers.png)
+
+<sup>figure 4: External Identity Provider Entity Relationship Diagram</sup>
+
+Entities holding configuration for external identity providers (google, github, facebook, etc) authenticating using Iridium. Other Iridium instances are registerable as external identity providers.
```
Comment (joshfischer1108) [inline site/docs/1.-Overview/iridium-database.md:59]: Google, Github, Facebook

**Review by joshfischer1108 — APPROVED** [review]
Really nice work @Horicabu .  I left a few comments, after those are addressed this is good to go.

💯

File: site/docs/1.-Overview/iridium-database.md (hunk 1 above)
Comment (Horicabu) [reply site/docs/1.-Overview/iridium-database.md]: done

File: site/docs/1.-Overview/iridium-database.md (hunk 3 above)
Comment (Horicabu) [reply site/docs/1.-Overview/iridium-database.md]: done

File: site/docs/1.-Overview/iridium-database.md (hunk 2 above)
Comment (Horicabu) [reply site/docs/1.-Overview/iridium-database.md]: done

**Comment by Horicabu** [comment]
@joshfischer1108 the fixes are addressed and committed.

**Review by joshfischer1108 — APPROVED** [review]
LGTM

**Pull request merged**